src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1624,6 +1634,20 @@ private void AssignPatternVariables(BoundPattern pattern, bool definitely = true
                                 AssignPatternVariables(sub.Pattern, definitely);
                             }
                         }
+                        if (pat.ListPatternClause is not null)
+                        {
+                            if (!pat.ListPatternClause.Subpatterns.IsDefaultOrEmpty)
+                            {
+                                foreach (BoundPattern p in pat.ListPatternClause.Subpatterns)
+                                {
+                                    AssignPatternVariables(p, definitely);
+                                }
+                            }
+                        }
+                        if (pat.LengthPattern is not null)
```

Comment:
nit: probably doesn't matter, but it feels like we should stick to the same order as the  syntax (ie. positional, then length pattern, then property/list clause)